Firestore/core/test/unit/model/value_util_test.cc

```diff
@@ -545,6 +545,12 @@ TEST_F(ValueUtilTest, DeepClone) {
   VerifyDeepClone(Map("a", Array("b", Map("c", GeoPoint(30, 60)))));
 }
 
+TEST_F(ValueUtilTest, CompareMapsWithDifferentKeyOrders) {
+  auto left = Map("a", 3, "b", 5);
+  auto right = Map("b", 5, "a", 3);
+  EXPECT_EQ(model::Compare(*left, *right), ComparisonResult::Same);
+}
```

Comment:
For completeness, please add a few more test cases where `left` and `right` have different conditions.

For example:
same keys and values with the same order
left and right have different number of keys
left and right have the same keys with different values